### DISCUSSION TITLE
There is a problem with the memory managed by MicroPython. How can I solve it?

### DISCUSSION THREAD

**Discussion by Mars-CN** [micropython/micropython#12373] [ESP32]
When I was porting MicroPython, I found that through gc.mem_ The memory obtained by free() is 3108592, gc.mem_ Alloc() is 19200, which means that the memory managed by MicroPython is only about 3M, but in reality, I am using ESP32-S3 N32R8V with 8M of memory. Where has the rest gone? How should I adjust this configuration?

**Comment by andrewleech** [comment]
Probably related to https://github.com/micropython/micropython/pull/12366

**Reply by projectgus** [reply]
Thanks Andrew, I think it may be that issue. 

(Otherwise, if @Mars-CN ported a version that's older than #12141 then MicroPython will use half the total available RAM for the Python heap (see #11853).)
